specification/automation/resource-manager/Microsoft.Automation/preview/2022-06-30-preview/dependency.json

```diff
@@ -0,0 +1,837 @@
+          "url": "http://aka.ms/azureautomationsdk/dependencyoperations"
+        },
+        "x-ms-examples": {
+          "Get a dependency": {
+            "$ref": "./examples/getDependency.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/AutomationAccountNameParameter"
+          },
+          {
+            "name": "runtimeName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of runtime."
+          },
+          {
+            "name": "dependencyName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The dependency name."
+          },
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/Dependency"
+            }
+          },
+          "default": {
+            "description": "Automation error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../common/v1/definitions.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "Dependency"
+        ],
+        "operationId": "Dependency_CreateOrUpdate",
+        "description": "Create or Update the dependency identified by name.",
+        "externalDocs": {
+          "url": "http://aka.ms/azureautomationsdk/dependencyoperations"
+        },
+        "x-ms-examples": {
+          "Create or update a dependency": {
+            "$ref": "./examples/createOrUpdateDependency.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../common/v1/definitions.json#/parameters/AutomationAccountNameParameter"
+          },
+          {
+            "name": "runtimeName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of runtime."
+          },
+          {
+            "name": "dependencyName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of dependency."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DependencyCreateOrUpdateParameters"
```

Comment:
>DependencyCreateOrUpdateParameters [](http://example.com/codeflow?start=37&length=34)

Your PUT body must be the same definition as your GET response, see here: [https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc008-put-get-patch--list-must-return-the-same-resource-schema](https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc008-put-get-patch--list-must-return-the-same-resource-schema)

Reply:
As discussed offline - For Dependency resource, we need file uploads (currently taking blob content uri) in PUT request, so couldn't make PUT contract same as GET contract. 
In this case, GET definition has more properties than the one from PUT, which is ok as RP is setting these additional properties which are readonly that the RP sets these values and are not meant to be set by customers.

So, PUT has one extra property for contentURI and all additional readonly properties in GET which are populated by RP are marked as readonly.